modules/virtual_network_gateway/main.tf

```diff
@@ -97,22 +97,25 @@ resource "azurerm_virtual_network_gateway" "this" {
 # https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip
 resource "azurerm_public_ip" "this" {
   for_each = { for ip_configuration in var.ip_configuration :
-    ip_configuration.name => try(ip_configuration.public_ip_standard_sku, false)
+    ip_configuration.name => {
+      name                   = ip_configuration.public_ip_name
+      public_ip_standard_sku = ip_configuration.public_ip_standard_sku
+    }
   if ip_configuration.create_public_ip }
 
   resource_group_name = var.resource_group_name
   location            = var.location
-  name                = "${var.name_prefix}pip-vgw${var.name_suffix}-${each.key}"
+  name                = each.value.name
 
-  allocation_method = each.value ? "Static" : "Dynamic"
-  zones             = var.enable_zones ? var.avzones : null
+  allocation_method = each.value.public_ip_standard_sku ? "Static" : "Dynamic"
```

Comment:
Dynamic allocation has two properties:
1. it does not need NSG for incoming internet traffic
2. IP is being rotated when the service becomes offline

For VPN configs I would say we should only do Static + NSG - to be sure that the IP is always the same and to control the incoming traffic

Reply:
I propose to leave that flexibility to choose e.g. for the lab cannot we use dynamic ?

Reply:
According to internal discussion, I defined allocation method for public IP as `Static` and SKU `Standard`.